.github/workflows/main.yml

```diff
@@ -0,0 +1,211 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+  schedule:
+    - cron: '0 0 * * 0'
+
+jobs:
+  yarn_install:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18
+
+      - name: Detect Yarn cache dir
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn config get cacheFolder)"
+
+      - name: Restore Yarn packages
+        id: yarn-restore
+        uses: actions/cache/restore@v3
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      # Restore node modules only if the SHA is perfect; to avoid possibility of corruption.
+      - name: Restore node_modules
+        id: node_modules-restore
+        uses: actions/cache/restore@v3
+        with:
+          path: ./node_modules
+          key: ${{ runner.os }}-node-modules-${{ hashFiles('yarn.lock') }}
+
+      - name: Yarn install
+        if: steps.node_modules-restore.outputs.cache-hit != 'true'
+        run: yarn install
+
+      - name: Save Yarn packages
+        if: steps.node_modules-restore.outputs.cache-hit != 'true'
+        uses: actions/cache/save@v3
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
+
+      - name: Save node_modules
+        if: steps.node_modules-restore.outputs.cache-hit != 'true'
+        uses: actions/cache/save@v3
+        with:
+          path: ./node_modules
+          key: ${{ runner.os }}-node-modules-${{ hashFiles('yarn.lock') }}
+
+  test_components:
+    runs-on: ubuntu-latest
+
+    needs: [yarn_install]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          # The storybook runner is going to do some Git operations that rely on access to the main branch.
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Setup Node
+        uses: getappmap/appmap-js@actions/setup-node/v1
```

Comment:
You can also [put the action in `.github/actions` directory](https://docs.github.com/en/actions/creating-actions/about-custom-actions#choosing-a-location-for-your-action).

Reply:
Updated. I don't know how you knew thish, I saw no reference to this anywhere...

Reply:
That must literally be the only place they mention that you can do that.

Reply:
> Updated. I don't know how you knew this, I saw no reference to this anywhere in GitHub docs or elsewhere...

Good google-fu xD